### ISSUE TITLE
Making DotNet Debugger a standalone project

### ISSUE THREAD

**Issue by enkomio** [0xd4d/dnSpy#1012]
This awesome project contains a lot of components that can be very helpful also by using as external library.

One of these is the managed debugger (which is, if I'm not wrong, the code from dnSpy.Debugger.DotNet.CorDebug).

Actually the alternatives are not very good, ClrMD has some issue when using the debugger interface (see issue 129). The best alternative is to use the CLR Managed Debugger (mdbg) Sample 4.0 from MS, which is not more maintained (last release is from 2016). It also has some bugs (it is not possible to set a breakpoint on overloading methods).

I'm not sure how complex could be this refactoring, but I think it would be very helpful :)

**Comment by 0xd4d** [comment]
Yes I've been thinking about that but haven't done it yet because

- Very few people would need such a library
- It's probably a lot of work

I'll keep it open but I have no idea if I'll ever create a new debugger lib.
